pkg/controllers/cloud_config_sync_controller_test.go

```diff
@@ -417,6 +444,14 @@ var _ = Describe("Cloud config sync reconciler", func() {
 			apierrors.IsNotFound(cl.Get(ctx, client.ObjectKeyFromObject(infra), infra)),
 		).Should(BeTrue())
 
+		networkResource := makeNetworkResource()
+		Expect(cl.Delete(ctx, networkResource)).To(Succeed())
+
+		Eventually(func() bool {
+			err := cl.Get(ctx, client.ObjectKeyFromObject(networkResource), networkResource)
+			return apierrors.IsNotFound(err)
+		}).Should(BeTrue())
```

Comment:
As above, True/False assertions can be hard to understand when they fail, if you want to do this, add context to the assertion.

An even better solution would be to match the error
```
Eventually(func() err {
			return cl.Get(ctx, client.ObjectKeyFromObject(networkResource), networkResource)
		}).Should(MatchError(apierrors.NewNotFound(config.Resource("networks"), networkResource.Name)))
```